### PR TITLE
avm2: Stub undocumented XML notification functions

### PR DESCRIPTION
**Pull request by evilpie**
Progresses https://github.com/ruffle-rs/ruffle/issues/12054. These function don't have `prototype` variants.

### CODE REVIEW

**Comment by evilpie** [comment]
Judging from the error stack trace I've seen, [mx.utils.XMLNotifier](https://help.adobe.com/en_US/FlashPlatform/reference/actionscript/3/mx/utils/XMLNotifier.html) must be using this under the hood.

File: core/src/avm2/globals/xml.rs [hunk 1]
```diff
@@ -759,3 +759,29 @@ pub fn replace<'gc>(
...
+
+pub fn set_notification<'gc>(
+    activation: &mut Activation<'_, 'gc>,
+    this: Object<'gc>,
+    args: &[Value<'gc>],
+) -> Result<Value<'gc>, Error<'gc>> {
+    avm2_stub_method!(activation, "XML", "setNotification");
+    let xml = this.as_xml_object().unwrap();
+    let node = xml.node();
+    let fun = args[0];
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/xml.rs:771]: This should probably be
```
let fun = args.try_get_object(activation, 0);
```
which reduces the following code to 
```
    node.set_notification(
        fun.as_function_object(),
        activation.context.gc_context,
    );
```

Comment (evilpie) [reply core/src/avm2/globals/xml.rs]: Nice, thanks.

**Review by Aaron1011 — APPROVED** [review]

File: core/src/avm2/globals/XML.as [hunk 2]
```diff
@@ -96,6 +96,10 @@ package {
             return this.toJSON(k);
         }
 
+        // undocumented functions
+        AS3 native function notification():Function;
+        AS3 native function setNotification(f:Function);
```
Comment (Lord-McSweeney) [inline core/src/avm2/globals/XML.as:101]: Playerglobal signature says that this returns `:*`- can you add it here?

**Review by Lord-McSweeney — APPROVED** [review]

**Review by sleepycatcoding — APPROVED** [review]

**Pull request merged**